src/course-outline/configure-modal/VisibilityTab.jsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Alert, Form } from '@edx/paragon';
+import { FormattedMessage, injectIntl } from '@edx/frontend-platform/i18n';
+import messages from './messages';
+
+const VisibilityTab = ({ isVisibleToStaffOnly, setIsVisibleToStaffOnly, showWarning }) => {
+  const handleChange = (e) => {
+    setIsVisibleToStaffOnly(e.target.checked);
+  };
+
+  return (
+    <>
+      <h3 className="mt-3"><FormattedMessage {...messages.sectionVisibility} /></h3>
+      <hr />
+      <Form.Checkbox checked={isVisibleToStaffOnly} onChange={handleChange}>
+        <FormattedMessage {...messages.hideFromLearners} />
+      </Form.Checkbox>
+      <hr />
+      {showWarning && (
```

Comment:
[In the spec](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3862560783/WBS+for+Course+Content+Outline+page#Content:-Section-Configure-(FE)), it seems that rendering this not-live is unintended, and instead can be derived from the current state of the checkbox.

> if checked: displays message (strange legacy behaviour: warning appears only on following renders only )

Reply:
Thank you for mentioning this and reminding me about it because I wanted to bring it up. 

In the legacy behavior, the warning is displayed if the section is already hidden from learners. We warn the cms user that some previously hidden content will be displayed. This seems sensible to me.

Toggling the warning based on the state of the checkbox feels somewhat off. Especially if the initial state is the content being not hidden. I thought we could maybe raise this point. What do you think?

Reply:
@CefBoud You are right, it makes sense to only show the warning if the content was already hidden.
cc: @mavidser